src/main/java/io/excaliburfrc/robot/subsystems/Climber.java

```diff
@@ -39,32 +85,86 @@ public void close() {
     }
   }
 
-  public void activateLeader(MotorMode m) {
-    motor.set(m.dutyCycle);
+  public void activateMotors(MotorMode m) {
+    leftMotor.set(m.dutyCycle);
+    rightMotor.set(m.dutyCycle);
   }
 
   public void openAngler() {
-    anglerPiston.set(DoubleSolenoid.Value.kForward);
+    piston.set(DoubleSolenoid.Value.kForward);
   }
 
   public void closeAngler() {
-    anglerPiston.set(DoubleSolenoid.Value.kReverse);
+    piston.set(DoubleSolenoid.Value.kReverse);
+  }
+
+  private Command reachSideHeight(
+      SparkMaxPIDController controller,
+      RelativeEncoder encoder,
+      double height,
+      double mg,
+      double angle) {
+    trapezoidGoal.position = height;
+    trapezoidInit.position = encoder.getPosition();
+    feedforward = new ElevatorFeedforward(0, mg * Math.cos(angle), kV, kA);
+    return new TrapezoidProfileCommand(
+            new TrapezoidProfile(
+                new TrapezoidProfile.Constraints(
+                    feedforward.maxAchievableVelocity(12, kA),
+                    feedforward.maxAchievableAcceleration(12, kV)),
+                trapezoidGoal,
+                trapezoidInit),
+            velocitySetpoint ->
+                controller.setReference(
+                    height,
+                    ControlType.kPosition,
+                    0,
+                    feedforward.calculate(velocitySetpoint.velocity),
+                    SparkMaxPIDController.ArbFFUnits.kVoltage),
+            this)
+        .withInterrupt(
+            () -> {
+              double position = encoder.getPosition();
+              return abs(height - position) <= THRESHOLD;
+            })
+        .andThen(
+            new ConditionalCommand(
+                new InstantCommand(() -> reachSideHeight(controller, encoder, height, mg, angle)),
+                new InstantCommand(() -> controller.setReference(0, ControlType.kDutyCycle)),
+                () -> abs(height - encoder.getPosition()) <= THRESHOLD));
```

Comment:
again, we can fairly assume that we're at the end of the profile when the command ends. no need for this.

if we see later that this assumption is wrong, we can fix it then